components/calendar/calendar-header.component.ts

```diff
@@ -108,6 +110,16 @@ export class NzCalendarHeaderComponent implements OnInit {
     this.setUpMonths();
   }
 
+  ngOnChanges(changes: SimpleChanges): void {
+    if (changes['activeDate']) {
```

Comment:
```suggestion
    if (changes.activeDate) {
```

Reply:
Hey thanks for the second review, we can keep the code as is. because when we write the dot notation syntax, the javascript engine consider the brackets syntax. So when we wrtite the dot notation syntax, first the javascript engine migrate to braket syntax and then perform the operation. 

By writting directly with bracket syntax we gain in performance

Reply:
Also we don't have the `@Inputs` on `changes`, this will cause in eslint throwing a warning saying that `acitveDate` might not exists on `changes`.

Reply:
> Also we don't have the `@Inputs` on `changes`, this will result in eslint throwing an error:
> 
> ![image](https://private-user-images.githubusercontent.com/62149413/295526460-c9c6b54e-fcb0-40cf-968c-9785a0d65056.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDQ4ODE0NDAsIm5iZiI6MTcwNDg4MTE0MCwicGF0aCI6Ii82MjE0OTQxMy8yOTU1MjY0NjAtYzljNmI1NGUtZmNiMC00MGNmLTk2OGMtOTc4NWEwZDY1MDU2LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAxMTAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMTEwVDEwMDU0MFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTM5MzkwZDcyM2M0ODc3NmE3ZWY3OWYyYmZjZGNkMWE0N2FiMGFiZDdjNGYxZWM5MjczZWE0ZTBjMzBmMmFmYjAmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.T0qCC6VMP_P04O01ysm-r5tf21wwAViP0yeovY9S8Nc)

also true :)

Reply:
Sorry, I didn't notice there were no `@Input`s, I just assumed the code should have been consistent with other components.

Anyway, I also suggested to fix a typo: `previuosActiveDate` -> `previousActiveDate`.

Reply:
no problem,
yeah you are right, I will fix that typo
@tegola

Reply:
@tegola @Nicoss54 
I forgot about this review; it is fixed now.